### PR TITLE
Contracts header with assertion macros specific to Sparrow

### PR DESCRIPTION
**Pull request by Klaim**
This provides macros (extracted from my library NoContracts) for contract assertions checks, named with SPARROW_ so that control over enabling these assertions is let to the user.

The checks are enabled by default and all C `assert(...)` calls have been replaced.

### CODE REVIEW

**Review by JohanMabille — COMMENTED** [review]
Same remarl as Alexis, I think `SPARROW_ASSERT_TRUE(!expr)` should be replaced with `SPARROW_ASSERT_FALSE`.

**Comment by Klaim** [comment]
OK I made a big replace operation to do that so now I'll review each assert see if the other option isnt better 👍🏽

**Review by Alex-PLACET — APPROVED** [review]
If you have the reference, I can say:
"Je contracte!"

File: include/sparrow/buffer.hpp
Comment (jjerphan) [inline include/sparrow/buffer.hpp:1]: Just a last comment: this file needs formatting and a few other ones as well it seems.

**Review by JohanMabille — APPROVED** [review]

File: include/sparrow/buffer.hpp
Comment (JohanMabille) [reply include/sparrow/buffer.hpp:1]: The current formatting rules mess the macro definitions, making them very hard to read. We should update the rules in a dedicated PR and then format.

**Pull request merged**